packages/flutter/lib/src/widgets/banner.dart

```diff
@@ -144,8 +153,8 @@ class BannerPainter extends CustomPainter {
       ..drawRect(_kRect, _paintShadow)
       ..drawRect(_kRect, _paintBanner);
     const double width = _kOffset * 2.0;
-    _textPainter.layout(minWidth: width, maxWidth: width);
-    _textPainter.paint(canvas, _kRect.topLeft + Offset(0.0, (_kRect.height - _textPainter.height) / 2.0));
+    _textPainter!.layout(minWidth: width, maxWidth: width);
```

Comment:
Developers are already leaking text painters today, they just have no way of disposing them. We should send out an announcement though and I can look into doing a g3 CL.

After dispose, the object should not be used - I don't think resetting prepared makes sense there, but we could add asserts about making sure methods aren't called after disposal.

And having a non-nullable but disposable object is often a little tricky, it's better to be able to null it out to prevent usage after disposal.

Reply:
It seems paragraphs can get different disposal signals:

- Leak: the developer doesn't call `dispose`, so a live paragraph is GC'd and `FinalizationRegistry` is pinged about it.
- Normal: developer calls `dispose`, then the GC collects the Dart wrapper and pings the `FinalizationRegistry` about it.

Do our tests cover both situations?

Reply:
We don't yet have a good way to cover an object that gets finalized without getting disposed, but @polina-c is working on that.